### PR TITLE
backend: Initialize Issues Controller (LabelLab 4)

### PR DESCRIPTION
**Pull request by M-A-D-A-R-A**
# Description

Initialize Issues Controller
Created a Controller to fetch all issue for the project 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

URL and Responses:- 
 - Create New Issue:- `URL:- BASE_URL +api/v1/issue/create/<int:project_id>`
     
  - Fetch all Issue in a project :- `URL:- BASE_URL +/api/v1/issue/<int:project_id>`
     
 - Get Issue by ID:- `URL:- BASE_URL +/api/v1/issue/issue_info/<int:project_id>/<int:issue_id>`
     
 - Update Issue by ID: -`URL:- BASE_URL +/api/v1/issue/issue_info/<int:project_id>/<int:issue_id>`
    
 - Delete issue by ID:- `URL:- BASE_URL +/api/v1/issue/issue_info/<int:project_id>/<int:issue_id>`
   
 - Fetch issue by Category :- `URL:- BASE_URL +/api/v1/issue/<int:project_id>/category/<string:category>`
    
 - Fetch issue by Team :- `URL:- BASE_URL +/api/v1/issue/<int:project_id>/team/<string:team>`
    

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by amlannandy — COMMENTED** [review]
I meant that you have to create all routes inside the controller. Not just one. And you only have to return a mock response inside each route. No need to do exception handling etc. 

Something like this.

```
class GetAllIssues(MethodView):
    """
    This class-based view handles fetching all issues within a project
    Url --> /api/v1/issue/get/<int:project_id>
    """
    @jwt_required
    def get(self, project_id):
        response = {
            "success": True,
            "msg": "Get All Project Issues",
        }
        return make_response(jsonify(response)), 200
        
class GetIssue(MethodView):
    """
    This class-based view handles fetching an issue by its id
    Url --> /api/v1/issue/issue_info/<int:issue_id>
    """
    @jwt_required
    def get(self, issue_id):
        response = {
            "success": True,
            "msg": "Get Issue by Id",
        }
        return make_response(jsonify(response)), 200
```

**Review by amlannandy — CHANGES_REQUESTED** [review]
Also do attach screenshots of all route responses once you've implemented them

File: labellab-flask/api/controllers/issuecontroller.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            return make_response(jsonify(response)), 500
+
+class FetchTeamIssuesView(MethodView):
+
+    """
+    This route fetches all project issue of a particular category
+     URL:- /api/v1/issue/<int:project_id>/team/<string:team>
+    """
+    @jwt_required
+    def get(self,project_id, team):
```
Comment (amlannandy) [inline labellab-flask/api/controllers/issuecontroller.py:159]: It should be `/api/v1/issue/<int:project_id>/team/<int:team_id>`. 
Team id is the primary key here, not team.

Comment (amlannandy) [inline labellab-flask/api/controllers/issuecontroller.py:159]: Change the variable name also to `team_id`

File: labellab-flask/api/routes/issue.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+)
+
+issuesprint.add_url_rule(
+    "issue/<int:project_id>/category/<string:category>",
+    view_func=issuecontroller.issueController["fetch_category_issue"], 
+    methods=["GET"]
+)
+
+issuesprint.add_url_rule(
+    "issue/<int:project_id>/team/<string:team>",
```
Comment (amlannandy) [inline labellab-flask/api/routes/issue.py:32]: Same change here

File: labellab-flask/api/controllers/issuecontroller.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+                "success":False,
+                "msg": "Something went wrong!"
+                }
+            # Return a server error using the HTTP Error Code 500 (Internal
+            # Server Error)
+            return make_response(jsonify(response)), 500
+
+
+issueController = {
+    "createissue": CreateIssues.as_view("createissue"),
```
Comment (amlannandy) [inline labellab-flask/api/controllers/issuecontroller.py:180]: Should be `create_issue`. As we're following snake case.

File: labellab-flask/api/routes/issue.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+from flask import Blueprint
+
+from api.controllers import issuecontroller
+
+issuesprint = Blueprint("issues", __name__)
+
+issuesprint.add_url_rule(
+    "/issue/create/<int:project_id>",
+    view_func=issuecontroller.issueController["createissue"],
```
Comment (amlannandy) [inline labellab-flask/api/routes/issue.py:9]: Same change to snake case here

**Review by amlannandy — CHANGES_REQUESTED** [review]

File: labellab-flask/api/controllers/issuecontroller.py (hunk 3 above)
Comment (M-A-D-A-R-A) [reply labellab-flask/api/controllers/issuecontroller.py]: Actucally i took reference from the `createproject` it was the not in snake case i changed that too in my new commit. Please have a look

Comment (amlannandy) [reply labellab-flask/api/controllers/issuecontroller.py]: Gotcha. Thanks for changing the other one too.

**Review by amlannandy — APPROVED** [review]
NIce work 👍🏻

**Pull request merged**